### PR TITLE
Merge ROS docker

### PR DESCRIPTION
**Pull request by guillermo-sarabia**

### CODE REVIEW

File: ci/ROS/Dockerfile_test [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+# Klepsydra
+COPY . kpsr-core
+
+RUN cd /opt/kpsr-core \
+    && git submodule update --init \
+    && mkdir build \
+    && cd build \
+    && cmake -DKPSR_WITH_DOXYGEN=true -DKPSR_WITH_DDS=false -DKPSR_WITH_ZMQ=false\
+       -DKPSR_TEST_PERFORMANCE=true -DKPSR_WITH_SOCKET=true -DKPSR_WITH_YAML=true\
+       -DKPSR_WITH_CODE_METRICS=true -DCMAKE_PREFIX_PATH=/opt/klepsydra/thirdparties\
```
Comment (pablogh-2000) [inline ci/ROS/Dockerfile_test]: Remove this: ```-DCMAKE_PREFIX_PATH=/opt/klepsydra/thirdparties```

File: ci/ROS_16.04/Dockerfile_test [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# Klepsydra
+COPY . kpsr-core
+
+RUN cd /opt/kpsr-core \
+    && git submodule update --init \
+    && mkdir build \
+    && cd build \
+    && cmake -DKPSR_WITH_DOXYGEN=true -DKPSR_WITH_DDS=false -DKPSR_WITH_ZMQ=false\
+       -DKPSR_TEST_PERFORMANCE=true -DKPSR_WITH_SOCKET=true -DKPSR_WITH_YAML=true\
+       -DKPSR_WITH_CODE_METRICS=true -DCMAKE_PREFIX_PATH=/opt/klepsydra/thirdparties\
```
Comment (pablogh-2000) [inline ci/ROS_16.04/Dockerfile_test]: Remove this ```-DCMAKE_PREFIX_PATH=/opt/klepsydra/thirdparties```

File: ci/Vanilla/Dockerfile_test [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# Klepsydra
+COPY . kpsr-core
+
+RUN cd kpsr-core \
+    && git submodule update --init \
+    && mkdir build \
+    && cd build \
+    && cmake -DKPSR_WITH_DOXYGEN=true -DKPSR_WITH_DDS=false -DKPSR_WITH_ZMQ=false\
+       -DKPSR_TEST_PERFORMANCE=true -DKPSR_WITH_SOCKET=true -DKPSR_WITH_YAML=false\
+       -DKPSR_WITH_CODE_METRICS=true -DCMAKE_PREFIX_PATH=/opt/klepsydra/thirdparties\
```
Comment (pablogh-2000) [inline ci/Vanilla/Dockerfile_test]: Remove this ```-DCMAKE_PREFIX_PATH=/opt/klepsydra/thirdparties```

File: ci/System_Dependencies/Dockerfile_ROS_16_04 [hunk 4]
```diff
@@ -0,0 +1,12 @@
+FROM osrf/ros:kinetic-desktop-full-xenial
+LABEL system_dependency=ubuntu_ROS:16.04
+
+# System Dependencies.
+RUN apt update && apt-get install libssl-dev libcurl4-gnutls-dev build-essential git cmake python cppcheck doxygen python3-pip gcovr lcov curl -y --fix-missing
```
Comment (pablogh-2000) [inline ci/System_Dependencies/Dockerfile_ROS_16_04]: Remove this ```libssl-dev libcurl4-gnutls-dev```

File: ci/System_Dependencies/Dockerfile_ROS_18_04 [hunk 5]
```diff
@@ -0,0 +1,12 @@
+FROM osrf/ros:melodic-desktop-full-bionic
+LABEL system_dependency=ubuntu_ROS:18.04
+
+# System Dependencies.
+RUN apt update && apt-get install libssl-dev libcurl4-gnutls-dev build-essential git cmake python cppcheck doxygen python3-pip gcovr lcov curl -y --fix-missing
```
Comment (pablogh-2000) [inline ci/System_Dependencies/Dockerfile_ROS_18_04]: Remove ```libssl-dev libcurl4-gnutls-dev```

File: ci/System_Dependencies/Dockerfile_Ubuntu_18_04 [hunk 6]
```diff
@@ -0,0 +1,9 @@
+FROM ubuntu:18.04
+LABEL system_dependency=ubuntu:18.04
+
+# System Dependencies.
+RUN apt update && apt-get install libssl-dev libcurl4-gnutls-dev build-essential git cmake python cppcheck doxygen python3-pip gcovr lcov curl -y --fix-missing
```
Comment (pablogh-2000) [inline ci/System_Dependencies/Dockerfile_Ubuntu_18_04]: Remove this ```libssl-dev libcurl4-gnutls-dev```

File: ci/ROS/Dockerfile_test (hunk 1 above)
Comment (guillermo-sarabia) [reply ci/ROS/Dockerfile_test]:
```suggestion
       -DKPSR_WITH_CODE_METRICS=true
```

File: ci/ROS_16.04/Dockerfile_test (hunk 2 above)
Comment (guillermo-sarabia) [reply ci/ROS_16.04/Dockerfile_test]:
```suggestion
       -DKPSR_WITH_CODE_METRICS=true \
```

**Pull request merged**